src/components/stackAggregator/index.js

```diff
@@ -101,55 +114,81 @@ export default class StackAggregator extends PureBaseComponent {
     return 1;
   }
 
-  animate = () => {
-    this.animateValues();
-    this.animateCards();
+  animate = async () => {
+    return Promise.all([this.animateValues(), this.animateCards()]);
   }
 
   animateValues() {
     const {collapsed} = this.state;
     const newValue = collapsed ? buttonStartValue : 1;
-
-    Animated.parallel([
-      Animated.timing(this.animatedOpacity, {
-        duration: DURATION,
-        toValue: Number(newValue),
-        useNativeDriver: true
-      }),
-      Animated.timing(this.animatedScale, {
-        toValue: Number(newValue),
-        easing: this.easeOut,
-        duration: DURATION,
-        useNativeDriver: true
-      }),
-      Animated.timing(this.animatedContentOpacity, {
-        toValue: Number(collapsed ? 0 : 1),
-        easing: this.easeOut,
-        duration: DURATION,
-        useNativeDriver: true
-      })
-    ]).start();
+    return new Promise((resolve) => {
+      Animated.parallel([
+        Animated.timing(this.animatedOpacity, {
+          duration: DURATION,
+          toValue: Number(newValue),
+          useNativeDriver: true
+        }),
+        Animated.timing(this.animatedScale, {
+          toValue: Number(newValue),
+          easing: this.easeOut,
+          duration: DURATION,
+          useNativeDriver: true
+        }),
+        Animated.timing(this.animatedContentOpacity, {
+          toValue: Number(collapsed ? 0 : 1),
+          easing: this.easeOut,
+          duration: DURATION,
+          useNativeDriver: true
+        })
+      ]).start(resolve);
+    })
   }
 
   animateCards() {    
+    const promises = []
     for (let index = 0; index < this.itemsCount; index++) {
       const newScale = this.getItemScale(index);
 
-      Animated.timing(this.animatedScaleArray[index], {
-        toValue: Number(newScale),
-        easing: this.easeOut,
-        duration: DURATION,
-        useNativeDriver: true
-      }).start();
+      promises.push(
+        new Promise((resolve) => {
+          Animated.timing(this.animatedScaleArray[index], {
+            toValue: Number(newScale),
+            easing: this.easeOut,
+            duration: DURATION,
+            useNativeDriver: true
+          }).start(resolve)
+        })
+      )  
     }
+    return Promise.all(promises)
   }
 
   close = () => {
-    this.setState({collapsed: true}, () => this.animate());
+    this.setState({collapsed: true}, async () => {
+      if (this.props.onCollapseWillChange) {
```

Comment:
You can use lodash's 'invoke' as a safe way to invoke a callback from props.

Reply:
gotcha. changed